malcolm/modules/ADPandABlocks/parts/kinematicssavupart.py

```diff
@@ -0,0 +1,395 @@
+                        dataset_i = ind
+                        self.use_min_max = False
+
+            # Check there was a dataset for the axis
+            assert dataset_i, "No value dataset for %s" % scannable
+	
+        if self.use_min_max:
+            dtypes["min"] = scanning.infos.DatasetType.POSITION_MIN
+            dtypes["max"] = scanning.infos.DatasetType.POSITION_MAX
+        for axis in self.q_value_mapping.values():
+            rank = len(generator.dimensions)
+            for k, v in dtypes.items():   
+                PATH='/entry/' + axis + "." + k
+                produced_datasets += [
+                    scanning.infos.DatasetProducedInfo(
+                                                        axis + "." + k,
+                                                        vds_fileName, v,
+                                                        rank, PATH, ""
+                                                      )
+                                     ]
+
+                                     
+        return produced_datasets        
+            
+
+    @add_call_types
+    def post_configure(self, context, part_info):
+        # type: (scanning.hooks.AContext, scanning.hooks.APartInfo) -> None
+
+        # Get the axis number for the inverse kinematics mapped in this cs_port
+        self.axis_numbers = pmac.util.cs_axis_numbers(
+            context, self.layout_table, self.cs_port
+        )
+
+        # Map these in the file
+        dataset_infos = scanning.infos.DatasetProducedInfo.filter_values(
+            part_info
+        )
+        for scannable, axis_num in self.axis_numbers.items():
+            min_i, max_i, value_i = None, None, None
+            for info in dataset_infos:
+                if info.name.startswith(scannable + "."):
+                    if info.type == scanning.infos.DatasetType.POSITION_MIN:
+                        min_i = info
+                    elif info.type == scanning.infos.DatasetType.POSITION_MAX:
+                        max_i = info
+                    elif info.type == scanning.infos.DatasetType.POSITION_VALUE:
+                        value_i = info
+            # Always make sure .value is there
+            assert value_i, "No value dataset for %s" % scannable
+            self.p_vars.append(PVar(
+                path=value_i.path, file=value_i.filename,
+                p_number="p%dmean" % axis_num)
+            )
+            if min_i and max_i:
+                self.p_vars.append(
+                    PVar(
+                        path=min_i.path, file=min_i.filename,
+                        p_number="p%dmin" % axis_num
+                    )
+                )
+                self.p_vars.append(PVar(path=max_i.path, file=max_i.filename,
+                                        p_number="p%dmax" % axis_num))
+            else:
+                self.use_min_max = False
```

Comment:
I would move this code up into on_configure, then you don't need to read the dataset infos that you just produced. Then all the min max parsing is in the same place

Reply:
I started attempting this change but then noted that we also use the file path and path within the file of the dataset in post_configure...can we reliably get this information without the datasetinfo?